src/services/room-service.js

```diff
@@ -140,13 +146,24 @@ export default class RoomService {
         joinedOn: new Date()
       };
 
-      await this.roomStore.updateOne(
-        { _id: invitation.roomId },
-        { $push: { members: newMember } },
-        { session }
-      );
+      let lock;
+
+      try {
+        lock = await this.roomLockStore.takeLock(invitation.roomId);
 
-      await this.roomInvitationStore.deleteOne({ _id: invitation._id }, { session });
+        const roomContainingNewMember = await this.roomStore.findOne({ '_id': invitation.roomId, 'members.userId': newMember.userId });
```

Comment:
You have to put the session also here as a param in order to get the data that is valid for this transaction

Reply:
Hmmm, okay, I thought since it's a read it doesn't matter. Fixed, thanks!